### PR TITLE
[v2.3.3] Vsphere Legacy Normalizer

### PR DESCRIPTION
**Pull request by luthermonson**
Adds a normalizer which updates legacy vSphere node templates to proper vmware object paths

### CODE REVIEW

File: pkg/controllers/management/nodetemplate/nodetemplate.go [hunk 1]
```diff
@@ -221,10 +223,44 @@ func (nt *nodeTemplateController) createGlobalNodeTemplateClone(legacyName, clon
...
+/*
+	legacy vmwarevsphere nodetemplates were free form text fields
+	and this is an attempt to normalize the data to a valid vsphere path
+	e.g. "My Network" becomes "/DC_NAME/networks/My Network"
+*/
+func vsphereLegacyNormalizer(nt *unstructured.Unstructured) {
+	k := vmwaredriver + "Config"
+	c := nt.Object[k].(map[string]interface{})
+
+	dc := c["datacenter"].(string)
```
Comment (rmweir) [inline pkg/controllers/management/nodetemplate/nodetemplate.go]: dc, _ := will avoid panic

File: pkg/controllers/management/nodetemplate/nodetemplate.go [hunk 2]
```diff
@@ -221,10 +223,44 @@ func (nt *nodeTemplateController) createGlobalNodeTemplateClone(legacyName, clon
...
+	if dc != "" && !strings.HasPrefix(dc, "/") {
+		dc = "/" + dc
+		c["datacenter"] = dc
+	}
+
+	ds := c["datastore"].(string)
```
Comment (rmweir) [inline pkg/controllers/management/nodetemplate/nodetemplate.go]: ds, _

File: pkg/controllers/management/nodetemplate/nodetemplate.go (hunk 1 above)
Comment (luthermonson) [reply pkg/controllers/management/nodetemplate/nodetemplate.go]: ah yep.

File: pkg/controllers/management/nodetemplate/nodetemplate.go (hunk 2 above)
Comment (luthermonson) [reply pkg/controllers/management/nodetemplate/nodetemplate.go]: ah yep.

**Pull request merged**